challenges/lib/src/main/java/challenges/tree/BinaryTree.java

```diff
@@ -0,0 +1,71 @@
+package challenges.tree;
+
+import java.util.ArrayList;
+
+public class BinaryTree {
+public Node root;
+
+    public BinaryTree(Node root) {
+        this.root = root;
+    }
+    public BinaryTree() {
+
+    }
+    //iterative

```

Comment:
You might want to try doing each of these methods, but recursively. It's much more simple than it may seem!